enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,104 @@
+all environments have multicast available. This proposal seeks to rememdy that.
+
+## Motivation
+
+Currently, the Baremetal platform utilizes keepalived to maintain the VIPs necessary
+for deployment and operation. It presently does so via multicast. This requires a
+minimal amount of configuration, as the multicast peers do not need to be specified.
+However, there are likely to be environments where multicast has been disabled because
+of network policy. In these cases, it is necessary to utilized unicast addresses within
+keepalived. Rather than make this an optional configuration, it makes sense to make it
+the default since unicast will "just work" in all environments.
+
+The focus is enabling us to deploy on multicast restricted environments provided the
+customer has the right external DNS records, but it will solve the possible collisions
+when two or more clusters are on the same broadcast/multicast domain.
+
+### Goals
+
+* A deployment is possible without the need for multicast on a network.
+* No loss of functionality from the current multicast implementation.
+
+### Non-Goals
+
+* A switchable implementation. It does not need to be an installer choice.
+
+## Proposal
+
+To enable keepalived to communicate via unicast instead of multicast, and to handle
+the fact that the bootstrap IP is unknown to the masters.
+
+### Implementation Details/Notes/Constraints
+
+* The bootstrap IP is not known until the server is up. Therefore, a method for 
+  retrieving that node will need to be implemented. The current POC implementation
+  uses a simple TCP server, to be called via the API VIP, that returns the IP
+  address of the bootstrap host. This server will be run on the bootstrap host as
+  a sidecar on the keepalived pod.
+
+* The bootstrap keepalived pod previously used a (mostly) static configuration
+  template. The keepalived monitor needs to run on the bootstrap host now. The
+  same monitor will be used for both the bootstrap and the control plane.
+
+### Risks and Mitigations
+
+It is important to not destabilize our existing installation method, that uses
+multicast successfully. Full parity is expected before landing.
+
+## Design Details
+
+### Test Plan
+
+Because of the fundamental inclusion of this in Baremetal IPI installations,
+any breakage by CI before landing can probably be assumed to be caused by
+this work.
+
+## Implementation History
+
+A PR with implementation in baremetal-runtimecfg can be found here:
+
+* https://github.com/openshift/baremetal-runtimecfg/pull/47
+
+The Machine Config operator portion can be found here:
+
+* https://github.com/openshift/machine-config-operator/pull/1504
+
+## Drawbacks
+
+* Multicast is a simpler configuration with fewer moving parts. However, not every
+  environment allows multicast, whereas all environments would allow unicast.
+
+* The customer must keep the Etcd SRV record up to date in case they add/replace
```

Comment:
I think there is a plan to remove the dependency on etcd SRV records

Reply:
That's good to know. Do you have a pointer to a discussion on that?

Reply:
@crawford @hexfusion

Reply:
Yes, we are dropping this dep in 4.4. etcd no longer is dependent on SRV in 4.4+